### DISCUSSION TITLE
Extend oxalis-as4 for a Production Ready Access Point

### DISCUSSION THREAD

**Discussion by anudhij** [OxalisCommunity/Oxalis-AS4#225] [Q&A]
Hi all,

I have used oxalis-as4 via DockerFile to run the OpenPeppol Testbed successfully. I have now requested for a Production PKI Certificate with OpenPeppol.

- I am looking for some guide or references, for me to have a production ready AP setup. Specifically, API based sending/receiving capability via Supplier/Customer to AccessPoints, C1 (Supplier) to send C2 (AP) the documents via API and C3 (AP) to send the document to C4 (Receiver) via API/Message Queue. If you have some reference or guideline which I should refer to, that will be great. 

- Also, I want to understand, how Oxalis-AS4/Oxalis connects with SMP (is there a configuration specified since there are many SMPs)

- I want to run the oxalis-as4 in my local system (Mac) using Intellij Idea, test the changes I want to make to extend the functionality via APIs.

Any guidance or reference in the above regards will be of huge help. Thanks!!

**Comment by Robcio35** [comment]
Hi @anudhij 

Did you find a solution for your question ? I have exactly the same situation. I successfully passed Testbed. Now I have to add necessary modules allowing backends (C1 and C4) to send and receive messages from Oxalis AS4.

Can you share your experience with me ?

Regards,

**Comment by platomedical** [comment]
- I am in the same boat here. The docker image works well. Now, I am trying to figure out how to set up the production environment.
- The "inbound" files as saved in a folder in docker. Looking for a way to reliably pass this to an external program for processing, or push it to an S3 bucket.
